### PR TITLE
chore: Added extra-checks to prevent unchecked submits on mainnet and testnet

### LINKED ISSUE #12363: Check ledger-id before sending unchecked submit
Before sending an unchecked submit in `SubmissionManager.submit()`, we check if the profile is not `PROD`. As an additional security measure, we should also check if the configured ledger-id is not mainnet or testnet.

### PR DESCRIPTION
**Pull request by netopyr**
This PR adds extra checks to secure unchecked submits even more.

Fixes #12363

### CODE REVIEW

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/SubmissionManager.java [hunk 1]
```diff
@@ -130,8 +133,12 @@ public void submit(@NonNull final TransactionBody txBody, @NonNull final Bytes t
...
             // We do NOT allow this call in production!
             // check profile dynamically, this way we allow profile overriding in Hapi tests
-            final var hederaConfig = configProvider.getConfiguration().getConfigData(HederaConfig.class);
-            if (hederaConfig.activeProfile() == Profile.PROD) {
+            final var configuration = configProvider.getConfiguration();
+            final var hederaConfig = configuration.getConfigData(HederaConfig.class);
+            final var ledgerConfig = configuration.getConfigData(LedgerConfig.class);
+            if (hederaConfig.activeProfile() == Profile.PROD
+                    || MAIN_NET_LEDGER_ID.equals(ledgerConfig.id())
+                    || TEST_NET_LEDGER_ID.equals(ledgerConfig.id())) {
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/SubmissionManager.java:141]: We should not allow in previewnet also right ?

Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/SubmissionManager.java]: Added a check for previewnet

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM ! Thanks @netopyr

**Review by tinker-michaelj — APPROVED** [review]
LGTM, thx @netopyr !

**Review by SimiHunjan — APPROVED** [review]

**Pull request merged**